### DISCUSSION TITLE
原生SharedPreferences异常清除数据

### DISCUSSION THREAD

**Discussion by lh5061765** [Tencent/MMKV#685] [Q&A]
您好，最近我有一些安卓设备，设备在异常断电关机的情况下，安卓APP使用SharedPreferences的方式保存数据信息，但是在设备再次开机后，有几率发现SharedPreferences数据被全部清空了，不知道使用MMKV是否可以避免这种情况

Hello, I have some Android devices recently. When the device is powered off abnormally, the Android APP uses SharedPreferences to save data information, but after the device is turned on again, there is a chance that the SharedPreferences data has been cleared, I don’t know how to use it. Can MMKV avoid this situation

**Comment by lingol** [comment]
Generally speaking, the data consistency of MMKV is higher than SP. 
But there's no guarantee on the unexpected shutdown of devices.
